Add PHP 8.0 support

bfoosness opened the pull request:
|    Q          |   A
|-------------- | ------
| New Feature   | yes

### Description

Close #13

bfoosness committed:
Update composer.json and fix unit tests.

Signed-off-by: Brent Foosness <bfoosness@users.noreply.github.com>

bfoosness commented on test/Validator/AuthenticationTest.php:
```diff
@@ -211,12 +211,7 @@ public function testNoServiceThrowsRuntimeException()
-            'messageTemplates',
-            $validator
-        );
+        $this->assertGreaterThan(0, count($this->validator->getOption('messageTemplates')));
```
I welcome guidance on this change. What does the previous test confirm, and does this change suffice?

svycka replied:
maybe like this?
```suggestion
        $r = new ReflectionObject($validator);
        $p = $r->getProperty('messageTemplates');
        $p->setAccessible(true);
        $this->assertEquals($validator->getOption('messageTemplates'), $p->getValue($validator));
```

holtkamp commented:
@boesing this PR seems to be complete to satisfy https://github.com/laminas/laminas-authentication/issues/13, does it need additional effort/work?

bfoosness committed:
Fix failing tests due to PHPUnit upgrade.

Signed-off-by: Brent Foosness <bfoosness@users.noreply.github.com>

bfoosness commented:
The laminas-db dependency is no longer a blocker, but I see I have a couple of issues to fix on this one now. I'll look at those this morning.

weierophinney committed:
qa: update composer lock, and remove obsolete CI information

Signed-off-by: Matthew Weier O'Phinney <matthew@weierophinney.net>

weierophinney committed:
qa: restore original test assertions

This patch uses `ReflectionProperty` to grab an internal value in order to perform an assertion equivalent to one performed in the 2.7.x series.

Signed-off-by: Matthew Weier O'Phinney <matthew@weierophinney.net>

weierophinney committed:
qa: consistency in path resolution of .gitignore file

Signed-off-by: Matthew Weier O'Phinney <matthew@weierophinney.net>

weierophinney commented:
I've restored the test case to be equivalent to what it was previously (not that it's terribly useful), and rebased your branch off current 2.8.x (which adds our GHA workflow) - all looks good, so I'm merging!
